restcomm/restcomm.sms/src/main/java/org/restcomm/connect/sms/SmsService.java

```diff
@@ -254,45 +256,61 @@ public void run() {
+                    //TODO:do extensions check here too?
+                    ActorRef session = session(this.configuration, sourceOrganizationSid);
+
+                    // Create an SMS detail record.
+                    final Sid sid = Sid.generate(Sid.Type.SMS_MESSAGE);
+                    final SmsMessage.Builder builder = SmsMessage.builder();
+                    builder.setSid(sid);
+                    builder.setAccountSid(client.getAccountSid());
+                    builder.setApiVersion(client.getApiVersion());
+                    builder.setRecipient(toUser);
+                    builder.setSender(client.getLogin());
+                    builder.setBody(new String(request.getRawContent()));
+                    builder.setDirection(Direction.OUTBOUND_CALL);
+                    builder.setStatus(Status.RECEIVED);
+                    builder.setPrice(new BigDecimal("0.00"));
+                    // TODO implement currency property to be read from Configuration
+                    builder.setPriceUnit(Currency.getInstance("USD"));
+                    final StringBuilder buffer = new StringBuilder();
+                    buffer.append("/").append(client.getApiVersion()).append("/Accounts/");
+                    buffer.append(client.getAccountSid().toString()).append("/SMS/Messages/");
+                    buffer.append(sid.toString());
+                    final URI uri = URI.create(buffer.toString());
+                    builder.setUri(uri);
+                    final SmsMessage record = builder.build();
+                    final SmsMessagesDao messages = storage.getSmsMessagesDao();
+                    messages.addSmsMessage(record);
+                    // Store the sms record in the sms session.
+                    session.tell(new SmsSessionAttribute("record", record), self());
+                    // Send the SMS.
+                    TlvSet tlvSet = new TlvSet();
+                    final SmsSessionRequest sms = new SmsSessionRequest(client.getLogin(), toUser, new String(request.getRawContent()), request, tlvSet, null);
+                    monitoringService.tell(new TextMessage(((SipURI) request.getFrom().getURI()).getUser(), ((SipURI) request.getTo().getURI()).getUser(), TextMessage.SmsState.INBOUND_TO_PROXY_OUT), self);
+                    session.tell(sms, self());
+                } else {
+                    if (logger.isDebugEnabled()) {
+                        final String errMsg = "Outbound SMS from Client " + client.getFriendlyName() + " not Allowed";
+                        logger.debug(errMsg);
+                    }
+                    String errMsg = "Outbound SMS from Client: " + client.getFriendlyName()
+                            + " is not Allowed";
+                    sendNotification(errMsg, 11001, "warning", true);
+                    final SipServletResponse resp = request.createResponse(SC_FORBIDDEN, "Call not allowed");
+                    resp.send();
```

Comment:
where do we cover postAction in this execution point (for both allowed and notAllowed options)?

Reply:
@jaimecasero I missed that, I will fix this